bind/motor-mobile/bind.go

```diff
@@ -34,38 +35,67 @@ func Init(buf []byte) ([]byte, error) {
 		}
 		return json.Marshal(resp)
 	}
-	return nil, errors.New("Loading existing account not implemented")
+	return nil, errors.New("loading existing account not implemented")
 }
 
 func CreateAccount(buf []byte) ([]byte, error) {
 	if instance == nil {
 		return nil, errWalletNotExists
 	}
-	if res, err := instance.CreateAccount(buf); err == nil {
+	// decode request
+	var request apiv1.CreateAccountRequest
+	if err := json.Unmarshal(buf, &request); err != nil {
+		return nil, fmt.Errorf("unmarshal request: %s", err)
+	}
+
+	if res, err := instance.CreateAccount(request); err == nil {
 		return json.Marshal(res)
 	} else {
 		return nil, err
 	}
 }
 
 func Login(buf []byte) ([]byte, error) {
-  if instance == nil {
-    return nil, errWalletNotExists
-  }
-
-  if res, err := instance.Login(buf); err == nil {
-    return json.Marshal(res)
-  } else {
-    return nil, err
-  }
+	if instance == nil {
+		return nil, errWalletNotExists
+	}
+
+	// decode request
+	var request apiv1.LoginRequest
+	if err := json.Unmarshal(buf, &request); err != nil {
+		return nil, fmt.Errorf("error unmarshalling request: %s", err)
+	}
+
+	if res, err := instance.Login(request); err == nil {
+		return json.Marshal(res)
+	} else {
+		return nil, err
+	}
+}
+
+func CreateSchema(buf []byte) ([]byte, error) {
+	if instance == nil {
+		return nil, errWalletNotExists
+	}
+
+	var request apiv1.CreateSchemaRequest
+	if err := json.Unmarshal(buf, &request); err != nil {
+		return nil, fmt.Errorf("unmarshal request: %s", err)
+	}
+
+	if res, err := instance.CreateSchema(request); err == nil {
+		return json.Marshal(res)
+	} else {
+		return nil, err
+	}
 }
 
 // Address returns the address of the wallet.
 func Address() string {
 	if instance == nil {
 		return ""
 	}
-	addr, err := instance.Wallet.Address()
+	addr, err := instance.GetWallet().Address()
```

Comment:
we should split this into two calls and check for nil on the wallet return

Reply:
done